### PR TITLE
Complete: To-Do-List Project

### PR DESCRIPTION
**Pull request by imahnama**
## In this Project We: 
- Used Factories to generate objects for the project
- Used single responsibility rule of Javascript
- Add properties to the To-Do's eg title, description, date and priority
- Add style to enhance UI
- Use localStorage to save user’s projects and todos between sessions.
- Use webpack
- Add ReadMe file
- Add linter files.

### CODE REVIEW

File: src/to-do.js [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    const toDoForm = document.createElement('div');
+    toDoForm.setAttribute('class', 'card project-module');
+    toDoForm.setAttribute('id', 'todoDoForm');
+    const title = document.createElement('label');
+    title.setAttribute('class', 'nameTxt');
+    toDoForm.appendChild(title);
+    title.innerHTML = 'Title';
+    const titleInput = document.createElement('input');
+    titleInput.setAttribute('id', 'title-id');
+    titleInput.type = 'text';
```
Comment (suretrust) [inline src/to-do.js:139]: You should separate your application logic (i.e. creating new todos, setting todos as complete, changing todo priority, etc.) from the DOM-related stuff, so keep all of those things in separate modules.

**Review by suretrust — CHANGES_REQUESTED** [review]
Hi team, 👋 

### Great job thus far! 🙌 Please see my comments. 👇 

- [x] The `todo.js` and `allProjects.js` files need some work. DOM-Manipulation and Logic should be in separate modules ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/modules_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/todo-list))
* [x] There should be a 'default' project, it's missing, let's add it.

**KIndly submit another CRR once these issues have been solved.** _Happy coding!_ 🎉

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    },
+    "extends": ["airbnb-base"],
+    "rules": {
+      "no-shadow": "off",
+      "no-param-reassign": "off",
+      "eol-last": "off",
+      "arrow-parens": "off",
+      "max-len": "off",
+      "no-use-before-define": "off",
+      "import/no-cycle": "off"
```
Comment (Torres-ssf) [inline .eslintrc.json]: - [x] This file is different from the one provided by Microverse [here](https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json). We should not make changes in this file

File: package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "JS-To-Do-List",
+  "version": "1.0.0",
+  "description": "Learning the concepts of Single Responsibility and Tightly Coupled Objects in Javascript",
+  "private": true,
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
+    "build": "webpack"
```
Comment (Torres-ssf) [inline package.json]: - [x] Great job implementing a script to use Webpack. However, it's best practice to create at least 2 scripts for your application, one for development and another one for production. We need to also to inform in which [mode](https://webpack.js.org/configuration/mode/) Webpack will run in these scripts. We will have something like this:
```javascript
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
  },
```

- [x] We also need to inform users about these script in the README file. This information can be placed in the Usage section. This way users will know how they can use your application.

**OPTIONAL**

- [x] You can create an even better script for development stages. You will need to add the `
webpack-dev-server` and create a script like this:
```javascript
"scripts": {
    "dev": "webpack --mode development",
    "build": "webpack --mode production",
    "start": "webpack-dev-server --mode development --open" // optional
  },
```
Running the `start` script will automatically open the index file in the browser, without the need to open it manually

File: package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "JS-To-Do-List",
+  "version": "1.0.0",
+  "description": "Learning the concepts of Single Responsibility and Tightly Coupled Objects in Javascript",
+  "private": true,
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1",
```
Comment (Torres-ssf) [inline package.json]: - [x] You can safely remove the test script since it has no purpose.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,60 @@
...
+- Bootstrap
+- Webpack
+
+## Live demo Link
+
+[live link](https://rawcdn.githack.com/OdongAlican/JS-To-Do-List/3a86b65dc66bbd7be54e098d4dd2f6db1ce57300/dist/index.html)
+
+![Landing page](1.png)
+![To-Do page](2.png)
+![Add To-Do Form](3.png)
```
Comment (Torres-ssf) [inline README.md]: Files `1.png`, `2.png`, and `3.png` are not in the root directory of your project. We need to specify the correct directory or they will not be displayed in the README file. Change this to:
```
(images/1.png)
(images/2.png)
(images/3.png)
```

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi team :hand: 
Great job on this project. Your UI it's really nice :clap: :clap: 
Nice job implementing the requested changes, but we still need to make other changes before moving on.

- [x] Check my comments bellow.

Feel free to message me on slack if you have any questions :smile: 
Happy Coding :computer: 
**Sergio Torres**

**Review by Oghenebrume50 — APPROVED** [review]
### APPROVED

Hi thanks for making changes as earlier requested by the reviewers, this looks good to me now, please merge and move on **This is approved** :+1: 

**RAPHAEL NORIODE** :mask: STAY_SAFE

**Pull request merged**